grails-app/controllers/org/broadinstitute/orsp/webservice/SampleCollectionController.groovy

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.orsp.webservice
+
+import com.google.gson.Gson
+import grails.converters.JSON
+import grails.rest.Resource
+import groovy.util.logging.Slf4j
+import org.broadinstitute.orsp.SampleCollection
+import org.springframework.web.multipart.MultipartFile
+
+import groovyx.net.http.HttpBuilder
+import org.broadinstitute.orsp.webservice.BspCollection
+
+@Slf4j
+@Resource(readOnly = false, formats = ['JSON', 'CSV'])
+class SampleCollectionController {
```

Comment:
My concern about this is that it is unauthenticated and therefore open to hacking. Since this is open source, it will be easy for anyone to POST to this API any kind of random data.

Reply:
Yes, I agree. I am looking to decide which approach to use.